ecmascript/parser/src/parser/class_and_fn.rs

```diff
@@ -45,31 +45,43 @@ impl<'a, I: Tokens> Parser<'a, I> {
         self.parse_fn(None, decorators)
     }
 
-    pub(super) fn parse_class_decl(&mut self, decorators: Vec<Decorator>) -> PResult<'a, Decl> {
-        self.parse_class(decorators)
+    pub(super) fn parse_class_decl(
+        &mut self,
+        start: BytePos,
```

Comment:
I copied what the rest of the code was doing by passing in byte positions, but I don't think this is ideal. I think it would be better to get the position within the functions then do assertions for... for example, functions like `parse_interface_decl` could start off with `let start = cur_pos!(); assert_and_bump!("interface")`. Anyway, doing that seemed like too big of a change for this PR.